docs/illink-options.md

```diff
@@ -0,0 +1,184 @@
+# IL Linker Command Line Options
+
+## illink tool
+
+The illink is IL Linker version shipping with .NET Core or .NET 5.0 platforms. It's bundled with
+the .NET SDK and most of the options are accessible using msbuild properties but any option
+can also be passed using `ExtraArgs` property.
+
+### Linking from the main assembly
+
+The command:
+
+`illink -a Program.exe`
+
+will use the assembly Program.exe as a source. That means that the linker will
+walk through all the methods of Program.exe to generate only what is necessary
+for this assembly to run.
+
+### Linking from an [XML descriptor](data-formats.md#descriptor-format)
+
+The command:
+
+`illink -x desc.xml`
+
+will use the XML descriptor as a source. That means that the linker will
+use this file to decide what to link in a set of assemblies. The format of the
+descriptors is described further on in this document.
+
+### Actions on the assemblies
+
+You can specify what the linker should do exactly per assembly.
+
+The linker can do the following things on all or individual assemblies
+
+- `skip` - skip them, and do nothing with them
+- `copy` - copy them to the output directory
+- `copyused` - copy used assemblies to the output directory
+- `link` - link them to reduce their size
+- `delete`- remove them from the output
+- `save` - save them in memory without linking
+- `AddBypassNGen` - add BypassNGenAttribute to unmarked methods
```

Comment:
IIRC you had some concerns about the `AddBypassNGen` attribute implementation - should we remove the documentation for them?

Reply:
Yeah they are a bit weird (removed them for now)